services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProviderFactory.java

```diff
@@ -164,6 +171,52 @@ public SAMLIdentityProviderConfig createConfig() {
                             samlIdentityProviderConfig.setEncryptionPublicKey(defaultCertificate);
                         }
                     }
+                    if (idpDescriptor.getExtensions() != null && idpDescriptor.getExtensions().getUIInfo() != null) {
+                        UIInfoType uiInfo = idpDescriptor.getExtensions().getUIInfo();
+                        // import attributes only if values with these locale exist
+                        uiInfo.getDisplayName().stream().filter(dn -> lang.equals(dn.getLang())).findFirst().ifPresent(
+                            displayName -> samlIdentityProviderConfig.setConfigMduiDisplayName(displayName.getValue()));
+                        uiInfo.getDescription().stream().filter(dn -> lang.equals(dn.getLang())).findFirst()
+                            .ifPresent(description -> samlIdentityProviderConfig.setMduiDescription(description.getValue()));
+                        uiInfo.getInformationURL().stream().filter(dn -> lang.equals(dn.getLang())).findFirst()
+                            .ifPresent(informationURL -> samlIdentityProviderConfig
+                                .setMduiInformationURL(informationURL.getValue().toString()));
+                        uiInfo.getPrivacyStatementURL().stream().filter(dn -> lang.equals(dn.getLang())).findFirst()
+                            .ifPresent(privacyStatementURL -> samlIdentityProviderConfig
+                                .setMduiPrivacyStatementURL(privacyStatementURL.getValue().toString()));
+                        if (!uiInfo.getLogo().isEmpty()) {
+                            LogoType logo = uiInfo.getLogo().get(0);
```

Comment:
`Logo` is a list of images and either should be used in full or not used at all.